src/app/components/project-map/project-map.component.ts

```diff
@@ -352,7 +356,23 @@ export class ProjectMapComponent implements OnInit, OnDestroy {
       });
   }
 
-  public onTextEdited(evt: TextEditedDataEvent){
+  public onTextAdded(evt: TextAddedDataEvent) {
```

Comment:
Probably we can use same approach like for drawing elements and move out this thing outside ProjectMap